instrumentation/github.com/labstack/echo/otelecho/test/echo_test.go

```diff
@@ -202,3 +200,57 @@ func TestErrorNotSwallowedByMiddleware(t *testing.T) {
 	err := h(c)
 	assert.Equal(t, assert.AnError, err)
 }
+
+func TestSpanNameFormatter(t *testing.T) {
+	imsb := tracetest.NewInMemoryExporter()
+	otel.SetTracerProvider(trace.NewTracerProvider(
+		trace.WithSyncer(imsb),
+	))
+
+	tests := []struct {
+		name      string
+		formatter otelecho.SpanNameFormatter
+		expected  string
+	}{
+		{
+			name:      "default",
+			formatter: nil,
+			expected:  "/user/:id",
+		},
+		{
+			name: "custom",
+			formatter: func(path string, req *http.Request) string {
+				return "custom " + path
+			},
+			expected: "custom /user/:id",
+		},
+		{
+			name: "has_request",
+			formatter: func(path string, req *http.Request) string {
+				return req.Method + " " + path
+			},
+			expected: "GET /user/:id",
```

Comment:
Just to preserve the original logic. But I also recommend this as the default.
I'm going to adjust this logic

Reply:
Then it seems that we just need to focus more on implementing method and target processing.
I will make adjustments later. and try to resolve the mistakes you raise.

Reply:
> I do not see a reason to add `SpanNameFormatter` config for `otelecho`. I added a comment in #6363.
> 
> At the same time, I see that the current span name is not following [open-telemetry/opentelemetry-specification@`v1.20.0`/specification/trace/semantic_conventions/http.md#name](https://github.com/open-telemetry/opentelemetry-specification/blob/v1.20.0/specification/trace/semantic_conventions/http.md#name) which is a separate bug.
> 
> I think that most (all?) HTTP instrumentation libraries may have this bug. _It would be good to double-check it and create bugs for tracking._

I adjusted according to the standards of the document.

Additionally, this will serve as a reference for other components. The adjustments for other components will be supplemented after this PR is approved.